tools/c7n_azure/c7n_azure/session.py

```diff
@@ -214,14 +215,20 @@ def client(self, client, vault_url=None):
         elif 'credentials' in klass_parameters:
             client = klass(credentials=self.credentials.legacy_credentials(self.resource_endpoint),
                            subscription_id=self.credentials.subscription_id,
-                           base_url=self.resource_endpoint)
+                           base_url=self.cloud_endpoints.endpoints.resource_manager)
             legacy = True
         else:
             client_args = {
                 'credential': self.credentials,
                 'raw_response_hook': log_response_data,
                 'retry_policy': C7nRetryPolicy()
             }
+
+            # This workaround will replace used api-version for costmanagement requests
```

Comment:
can we add a todo here, along with the current cost sdk version number, just to raise the visibility, ie eventually the cost sdk will get fixed..

Reply:
Added todo.. I will create c7n workitems for the remaining/outstanding work, so we can track it.